src/js/services/fs-xhr.js

```
@@ -42,11 +44,20 @@ define('services/fs-xhr',['q','jquery'],function(Q,$) {
         this._delete(url).done(function(data) {
             def.resolve(data);
         }).fail(function(data) {
+            handleFailedRequest(data);
             def.reject(data);
         });
         return def.promise;
     }
 
+    function handleFailedRequest(responseData) {
+        if (responseData.statusText === "error") { //status text given when server is inaccessible
+            alert(`Server is inaccessible!
```

alerting as a UI operation, and this files (this and ng-message) are logical services. please throw the error, or use promise's reject function, and deciding on the proper UI move in the controllers

---

I'm not sure this is a good idea. If we do that, this means that everywhere we use those two services we will need to also add this alert. To me, it seems logical that when a service doesn't work it alerts the uses it cannot work and why, as well as rejecting the promise or throwing an error.

---

You're right, but you need to remember that the user of this service is the controller, not the actual user. Yes, it means that everywhere we have to catch an error and display a message. Which means, if tomorrow we decide that there is no place for error messages only in the scoresheet, we can, because the controller is in charge of the UI. This separation between logic and UI is important for a steady architecture